skbio/metadata/_testing.py

```diff
@@ -904,3 +905,166 @@ def test_has_positional_metadata(self):
         obj = self._positional_metadata_constructor_(
             2, positional_metadata={'foo': [1, 2], 'bar': ['abc', 'def']})
         self.assertTrue(obj.has_positional_metadata())
+
+
+class IntervalMetadataMixinTests:
+    def _set_up(self):
+        self.upper_bound = 9
+        self.im = IntervalMetadata(self.upper_bound)
+        self.intvls = [
+            {'locations': [(0, 1)], 'metadata': {'gene': 'sagA'}},
+            # repeat it
+            {'locations': [(0, 1)], 'metadata': {'gene': ['a'],
+                                                 'product': 'foo'}}]
+
+    def test_constructor_invalid(self):
+        with self.assertRaises(TypeError):
+            self._interval_metadata_constructor_(0, '')
+
+    def test_eq_basic(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        im2 = IntervalMetadata(self.upper_bound)
+        im2.add(**self.intvls[0])
+        obj2 = self._interval_metadata_constructor_(self.upper_bound, im2)
+
+        self.assertReallyEqual(obj1, obj2)
```

Comment:
We should continue using `skbio.util._testing.ReallyEqualMixin` because it performs other equality sanity checks, such as making sure `A == B` and `B == A`, and `A != B` and `B != A`. This can catch subtle bugs in equality implementations. The mixin in scikit-bio doesn't include the `cmp` checks as those are no longer necessary in Python 3.